src/plugins/initiatives/initiativesDirectory.js

```diff
@@ -61,3 +70,31 @@ export function fromJSON(j: Compatible<any>): InitiativeFile {
 export function toJSON(m: InitiativeFile): Compatible<InitiativeFile> {
   return toCompat(COMPAT_INFO, m);
 }
+
+/**
+ * When provided with the initiative NodeAddressT of an InitiativeFile this extracts
+ * the URL from it. Or null when the address is not for an InitiativeFile.
+ */
+export function initiativeFileURL(address: NodeAddressT): string | null {
+  const initiativeFilePrefix = NodeAddress.append(
+    initiativeNodeType.prefix,
+    INITIATIVE_FILE_SUBTYPE
+  );
+
+  if (!NodeAddress.hasPrefix(address, initiativeFilePrefix)) {
+    return null;
+  }
```

Comment:
Can you explain why we want this behavior, rather than throwing an error when we got a non-initiative address?

Reply:
Not throwing makes it really easy to compose this function should you want to.
An example usage is in our own `initiatives/createGraph` https://github.com/sourcecred/sourcecred/commit/3749a422bda92a2d631b7fc48773079e9f8118f5 (prototype)

As this is kind of a "reverse" reference detector, that might be interesting outside this module too.
https://github.com/sourcecred/sourcecred/blob/f924521fdd0761f448bc7463fcf91bf1ec59b444/src/core/references/referenceDetector.js#L18

However you're right in the context of purely converting InitiativeFile to Initiative, this should throw. And we do, in the follow-up PR #1648.
https://github.com/sourcecred/sourcecred/blob/f867f5d9d8d55fcc567286cd67c0d13c555abf0e/src/plugins/initiatives/initiativesDirectory.js#L195-L198

Reply:
Cool, thanks. I love that we have Flow to ensure that all clients will need to handle the case where it doesn't correspond to an initiative.